dockerfiles/ci/buster/php-master/Dockerfile

```diff
@@ -10,10 +10,8 @@ RUN set -eux; \
 COPY install-php-master /usr/local/bin/
 RUN install-php-master
 
-# Install Composer
-RUN set -eux; \
-    curl -q https://raw.githubusercontent.com/composer/getcomposer.org/76a7060ccb93902cd7576b67264ad91c8a2700e2/web/installer | \
-    sudo php -- --filename=composer --install-dir=/usr/local/bin;
+
+COPY --from=composer:2 /usr/bin/composer /usr/local/bin/composer
```

Comment:
I do not see the advantage of doing this (TIL we do in other places, though).

When we do this we have to download the composer image in order only to extract a file (evidence of that)
``` => FROM docker.io/library/composer:2                                                                                                                                        14.5s
 => => resolve docker.io/library/composer:2                                                                                                                                   2.3s
 => => sha256:aca054f37f7b1a5f63bc8363a31c383ac9e2a4edd9983b92421c34efa1f35b2a 1.65kB / 1.65kB                                                                                0.0s
 => => sha256:743181516422399af3888ad5df44c6f711f992b2f86288db9e191ae13de99fa0 1.69MB / 1.69MB                                                                                0.4s
 => => sha256:e2d5b43308915ce3198dec1c19aa88719afacff02c85660a952982856a0bcc9c 268B / 268B                                                                                    0.5s
 => => sha256:6c307b60e377a3903f74fd9cde3ad69f754071faac36f81ca74e1aa9eabe4851 1.26kB / 1.26kB                                                                                0.5s
 => => sha256:ff854871a5957060e93934887728d773bd21351468fc6b4b6ac5465ad2c81ea2 13.32kB / 13.32kB                                                                              0.0s
 => => sha256:3ee24d70a08e170bdb914f6bc4cea57fee943ff43c6856a3f168ee993e53e786 3.46kB / 3.46kB                                                                                0.0s
 => => extracting sha256:743181516422399af3888ad5df44c6f711f992b2f86288db9e191ae13de99fa0                                                                                     0.2s
 => => sha256:0baf29266ab77eda55610c722f3b903364328c7c92117690eb657655e419bf9d 10.78MB / 10.78MB                                                                              4.4s
 => => sha256:dfa723e4109be76aefa6e8bc28bcacb55aa90d501f03ffc5649eebd131ceed09 493B / 493B                                                                                    0.7s
 => => sha256:7e9fbc8e5c1a07d8bdbc757bd388cbfe410d3eec2cb1ec49ea770320ae242dc7 14.95MB / 14.95MB                                                                              6.1s
 => => extracting sha256:6c307b60e377a3903f74fd9cde3ad69f754071faac36f81ca74e1aa9eabe4851                                                                                     0.0s
 => => sha256:510d7725cdd56d978d0dfb33354aea034a3da8931963851655e104f0e1db0463 2.27kB / 2.27kB                                                                                1.0s
 => => extracting sha256:e2d5b43308915ce3198dec1c19aa88719afacff02c85660a952982856a0bcc9c                                                                                     0.0s
 => => sha256:5fdc925b16f55706769b248f0cffef3239f6290a6ba31d73d3c1011723849d36 17.52kB / 17.52kB                                                                              1.4s
 => => sha256:c898ee3b4745b6a914bc0c00649d2e875d9a0affe2a2348cf3b40137347ea3a7 31.16MB / 31.16MB                                                                              6.0s
 => => sha256:f712c6bde093375cf23866a1ec9cc403af45a0b97dfa6194e82a788ae309c3b7 198.23kB / 198.23kB                                                                            4.8s
 => => extracting sha256:0baf29266ab77eda55610c722f3b903364328c7c92117690eb657655e419bf9d                                                                                     0.1s
 => => extracting sha256:dfa723e4109be76aefa6e8bc28bcacb55aa90d501f03ffc5649eebd131ceed09                                                                                     0.0s
 => => sha256:02b0dc7f879137009be18156044da5d8d5eb9b720fb2ac82021a3407d407c391 258B / 258B                                                                                    5.0s
 => => sha256:37eb1a7b3bf8c3a6cbe7b93e0a1bd979270105313f6dc3bfd6856b31f32f6367 554.12kB / 554.12kB                                                                            5.7s
 => => sha256:a53a2191b206d36a2c5b9109f227baae4f7f398253a18268080a66745a783c59 409B / 409B                                                                                    6.0s
 => => sha256:8f5c6ac570cfb9ffd956004be1ecd9bee6d150945b96c84252506fc2f779383f 126B / 126B                                                                                    6.2s
 => => extracting sha256:7e9fbc8e5c1a07d8bdbc757bd388cbfe410d3eec2cb1ec49ea770320ae242dc7                                                                                     1.8s
 => => extracting sha256:510d7725cdd56d978d0dfb33354aea034a3da8931963851655e104f0e1db0463                                                                                     0.0s
 => => extracting sha256:5fdc925b16f55706769b248f0cffef3239f6290a6ba31d73d3c1011723849d36                                                                                     0.0s
 => => extracting sha256:c898ee3b4745b6a914bc0c00649d2e875d9a0affe2a2348cf3b40137347ea3a7                                                                                     4.1s
 => => extracting sha256:f712c6bde093375cf23866a1ec9cc403af45a0b97dfa6194e82a788ae309c3b7                                                                                     0.2s
 => => extracting sha256:02b0dc7f879137009be18156044da5d8d5eb9b720fb2ac82021a3407d407c391                                                                                     0.0s
 => => extracting sha256:37eb1a7b3bf8c3a6cbe7b93e0a1bd979270105313f6dc3bfd6856b31f32f6367                                                                                     0.2s
 => => extracting sha256:a53a2191b206d36a2c5b9109f227baae4f7f398253a18268080a66745a783c59                                                                                     0.0s
 => => extracting sha256:8f5c6ac570cfb9ffd956004be1ecd9bee6d150945b96c84252506fc2f779383f                                                                                     0.0s
```

I do not find nice for who build the image to have to download extra GBs

Reply:
This was part of the cherry-picked commit (5ed52aa) from #1061. Using Composer 2 as part of any new container builds (e.g. `datadog/dd-trace-ci:php-8.0-shared-ext`) came as a special request from @morrisonlevi so I'll let him chime in here. :)

Reply:
I'm not sure I understand your concern, Luca. Since we are including composer in the image one way or another, we have to download it. All I did was change it from messy commands to a simple docker-idiomatic one.

So... maybe I didn't catch a subtlety or something?

Reply:
This `composer` layer gets added to more than one image, and I use it for other images I make for playing around as well, so for me at least it's well cached and well worth the re-use.

Reply:
I do not consider this a blocker. I am resolving this comment so we can unblock the PR.